kernel/byterun/coq_interp.c

```diff
@@ -16,17 +16,35 @@
 #include <stdio.h>
 #include <signal.h>
 #include <stdint.h>
+#include <math.h>
+
+#define CAML_INTERNALS
 #include <caml/memory.h>
 #include <caml/signals.h>
 #include <caml/version.h>
-#include <math.h>
-#include "coq_gc.h"
+
 #include "coq_instruct.h"
 #include "coq_fix_code.h"
 #include "coq_memory.h"
 #include "coq_values.h"
 #include "coq_float64.h"
 
+#if OCAML_VERSION < 41000
+#undef Alloc_small
+#define Alloc_small(result, wosize, tag) do{                            \
+  young_ptr -= Bhsize_wosize (wosize);                                  \
+  if (young_ptr < young_limit){                                         \
+    young_ptr += Bhsize_wosize (wosize);                                \
+    Setup_for_gc;                                                       \
+    minor_collection ();                                                \
```

Comment:
Better writing `caml_minor_collection` directly rather than relying on the compatibility macro.

Reply:
Idem for `caml_young_ptr`, `caml_young_limit`. Hint: pass `-DCAML_NAME_SPACE` on the command line to the C compiler.